fdbserver/MockPeekCursor.h

```diff
@@ -21,39 +21,102 @@
 #ifndef FDBSERVER_MOCK_PEEK_CURSOR
 #define FDBSERVER_MOCK_PEEK_CURSOR
 
+#include <ostream>
 #include "fdbserver/LogSystem.h"
 
+struct VersionedMessage {
+	// Every message should be able to have independent arenas so that they can be freed separately after consumed.
+	Arena arena;
+	Version version;
+	// Subversion is not used for now.
+	uint32_t sub = 0;
+	StringRef message;
+	VectorRef<Tag> tags{};
+	VersionedMessage() = default;
+	VersionedMessage(const Arena& arena, Version version, const StringRef& message, const VectorRef<Tag>& tags)
+	  : arena(arena), version(version), message(message), tags(tags) {}
+
+	friend std::ostream& operator<<(std::ostream& os, const VersionedMessage& versionedMessage) {
```

Comment:
nit: this function probably doesn't need to be inlined. If you wanted to make a little bit of extra effort to improve compile times, this header could include `<iosfwd>` instead of `<ostream>` and only include `<ostream>` in the cpp.